### PR TITLE
[BUG] potential typo in default learning rate

### PR DESCRIPTION
**Pull request by flennerhag**
Training any baseline model with default settings cause exploding loss or no learning at all. 

The default is set at 20. Setting it to something like 1 or 2 allows the models to learn.

### CODE REVIEW

**Comment by soumith** [comment]
the default setting is border-line fastest learning rate we can use, i believe.

**Comment by flennerhag** [comment]
Ok, I pushed this just in case it was a typo.

I suppose the default has been tested so should work. Currently I only have access to my laptop (mac), which might be a fringe case. In this case, running ``RNN_RELU`` with defaults arguments produces ``nan`` losses immediately. Running ``RNN_TANH`` with defaults causes training to plateau at a loss of roughly 10 (> 1000 in ppl). The losses I'm referring to are the printouts so the actual loss used by ``torch`` in the backprop step might not suffer from overflow, but regardless on my machine the loss is unstable and doesn't improve as expected.

Being examples, it might be an idea to lower the learning rate a bit to ensure training is stable, but that's of course up to you. If you're happy with the current default, feel free to close this PR.

**Comment by apaszke** [comment]
I'd add gradient clipping. Exploding gradients are common in RNNs, and would happen even with a smaller LR (unless you actually make it extremely small)

**Comment by apaszke** [comment]
I think we'll leave the defaults as they are. We're aware they won't work for all kinds of models, but that's how it is with hyperparameters.

**Comment by flennerhag** [comment]
Sorry forgot about this. Actually, I found that the problem has to do with calculating the ppl score for the batches early in the first epoch. If the loss starts out at or above ``700``, this will cause an ``OverflowError`` when calling ``math.exp`` that aborts the entire program.

Putting a ``try-except`` block around calculating the perplexity score (``math.exp(cur_loss)``) solved the issue. Let me know if you'd like me to fire up a PR for it.
